src/gui/transferlistsortmodel.cpp

```diff
@@ -180,85 +164,16 @@ bool TransferListSortModel::lessThan_impl(const QModelIndex &left, const QModelI
-
-            const TransferListModel *model = qobject_cast<TransferListModel *>(sourceModel());
-
-            // From QSortFilterProxyModel::lessThan() documentation:
-            //   "Note: The indices passed in correspond to the source model"
-            const bool isActiveL = TorrentFilter::ActiveTorrent.match(model->torrentHandle(left));
-            const bool isActiveR = TorrentFilter::ActiveTorrent.match(model->torrentHandle(right));
-            if (isActiveL != isActiveR)
-                return isActiveL;
-
-            const auto queuePosL = left.sibling(left.row(), TransferListModel::TR_QUEUE_POSITION)
-                    .data(TransferListModel::UnderlyingDataRole).toInt();
-            const auto queuePosR = right.sibling(right.row(), TransferListModel::TR_QUEUE_POSITION)
-                    .data(TransferListModel::UnderlyingDataRole).toInt();
-            const bool isSeedingL = (queuePosL < 0);
-            const bool isSeedingR = (queuePosR < 0);
-            if (isSeedingL != isSeedingR)
-            {
-                const bool isAscendingOrder = (sortOrder() == Qt::AscendingOrder);
-                if (isSeedingL)
-                    return !isAscendingOrder;
-
-                return isAscendingOrder;
-            }
-
-            const auto etaL = leftValue.toLongLong();
-            const auto etaR = rightValue.toLongLong();
-            const bool isInvalidL = ((etaL < 0) || (etaL >= MAX_ETA));
-            const bool isInvalidR = ((etaR < 0) || (etaR >= MAX_ETA));
-            if (isInvalidL && isInvalidR)
-            {
-                if (isSeedingL)  // Both seeding
-                    return invokeLessThanForColumn(TransferListModel::TR_SEED_DATE);
-
-                return (queuePosL < queuePosR);
-            }
-
-            if (!isInvalidL && !isInvalidR)
-                return (etaL < etaR);
-
-            return !isInvalidL;
-        }
-
-    case TransferListModel::TR_LAST_ACTIVITY:
-        {
-            const auto lastActivityL = leftValue.toLongLong();
-            const auto lastActivityR = rightValue.toLongLong();
-
-            if (lastActivityL < 0) return false;
-            if (lastActivityR < 0) return true;
-            return lastActivityL < lastActivityR;
-        }
-
-    case TransferListModel::TR_RATIO_LIMIT:
-        {
-            const auto ratioL = leftValue.toReal();
-            const auto ratioR = rightValue.toReal();
-
-            if (ratioL < 0) return false;
-            if (ratioR < 0) return true;
-            return ratioL < ratioR;
-        }
+    default:
+        Q_ASSERT_X(false, Q_FUNC_INFO, "Missing comparsion case");
+        break;
     }
 
-    return (leftValue != rightValue)
-        ? QSortFilterProxyModel::lessThan(left, right)
-        : invokeLessThanForColumn(TransferListModel::TR_QUEUE_POSITION);
+    // return `false` by default since `left` is not *strictly* less than `right`
+    return false;
```

Comment:
So it's still enough?

Reply:
Seems so. I dropped all position query code since the PR still works. IIRC I did try the same a long time ago and Qt at that time wouldn't do stable sort.